pkgs/tools/system/zenith/default.nix

```diff
@@ -15,12 +15,15 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs = stdenv.lib.optionals stdenv.isDarwin [ IOKit ];
 
+  preBuild = ''
+    export CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=gcc
```

Comment:
Instead of using `preBuild`, I believe you can simply write 
```
CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER = "gcc";
```
and the call to `mkDerivation` underlying `buildRustPackage` will interpret it as setting an environment variable for the build process.